### DISCUSSION TITLE
Supabase bug? or something with my code?

### DISCUSSION THREAD

**Discussion by oxrinz** [supabase/supabase#27635] [Questions] [answered]
```
Unhandled Runtime Error
TypeError: Cannot destructure property 'PostgrestClient' of '_cjs_index_js__WEBPACK_IMPORTED_MODULE_0___default(...)' as it is undefined.

Call Stack
eval
node_modules\@supabase\postgrest-js\dist\esm\wrapper.mjs (3:1)
(app-pages-browser)/./node_modules/@supabase/postgrest-js/dist/esm/wrapper.mjs
file:///D:/DOSE%202.0/site%204.0/dose-website-5.0/.next/static/chunks/app/account/page.js (325:1)
options.factory
file:///D:/DOSE%202.0/site%204.0/dose-website-5.0/.next/static/chunks/webpack.js (715:31)
Next.js
fn
file:///D:/DOSE%202.0/site%204.0/dose-website-5.0/.next/static/chunks/webpack.js (371:21)
eval
(app-pages-browser)/./node_modules/@supabase/supabase-js/dist/module/SupabaseClient.js
(app-pages-browser)/./node_modules/@supabase/supabase-js/dist/module/SupabaseClient.js
file:///D:/DOSE%202.0/site%204.0/dose-website-5.0/.next/static/chunks/app/account/page.js (654:1)
options.factory
file:///D:/DOSE%202.0/site%204.0/dose-website-5.0/.next/static/chunks/webpack.js (715:31)
Next.js
fn
file:///D:/DOSE%202.0/site%204.0/dose-website-5.0/.next/static/chunks/webpack.js (371:21)
eval
(app-pages-browser)/./node_modules/@supabase/supabase-js/dist/module/index.js
(app-pages-browser)/./node_modules/@supabase/supabase-js/dist/module/index.js
file:///D:/DOSE%202.0/site%204.0/dose-website-5.0/.next/static/chunks/app/account/page.js (665:1)
options.factory
file:///D:/DOSE%202.0/site%204.0/dose-website-5.0/.next/static/chunks/webpack.js (715:31)
Next.js
__webpack_require__
file:///D:/DOSE%202.0/site%204.0/dose-website-5.0/.next/static/chunks/webpack.js (37:33)
```

client.ts
```
import { createBrowserClient } from "@supabase/ssr";

export const createClient: any = () => {
    createBrowserClient(
        process.env.NEXT_PUBLIC_SUPABASE_URL!,
        process.env.NEXT_PUBLIC_SUPABASE_ANON_KEY!
    )
}
```

page.tsx
```
import { createClient } from '@/util/supabase/server'

export default async function Account() {
  const supabase = createClient()
}
```

The code is very minimal, has anyone run into this before, or should I create a bug report? Everything was working fine until I started working on auth and middleware, installed a few other supabase packages, unsure which and in which order, but after installing those it all broke. I tried reinstalling the packages and that didn't work. Might be something with package types, I'm not educated on those.

Any help appreciated

**Comment by z-x** [comment]
I just updated my packages and have the same issue without even touching the code. Everything is working fine before the updates.

![image](https://github.com/supabase/supabase/assets/1907443/3e68e2ea-5dad-4c87-9b5e-0978291b801d)

**Comment by bonchevskyi** [comment]
same issue for me when trying to run `yarn build`:
```ts
TypeError: Cannot destructure property 'PostgrestClient' of 'p.n(...)(...)' as it is undefined.
```
here is versions which I use:
```json
"packageManager": "yarn@4.3.1",
"next": "^14.2.4",
"react": "^18.3.1",
"@supabase/ssr": "^0.4.0",
"@supabase/supabase-js": "^2.44.2",
```

**Comment by chanceDira** [comment]
I have facing the same issue, I haven't changed any code

**Comment by richardnqoko** [comment]
I had the same issue but solved it by doing:

npm uninstall @supabase/supabase-js

npm install @supabase/supabase-js@2.43.4

**Reply by chanceDira** [reply]
> I had the same issue but solved it by doing:
> 
> npm uninstall @supabase/supabase-js
> 
> npm install @supabase/supabase-js@2.43.4

@richardnqoko I tried it but I am still getting the same issue

**Reply by chanceDira** [reply]
@richardnqoko wait seems to work, thank you

**Reply by GyoHeon** [reply]
@richardnqoko It works to me, THX!

**Reply by oxrinz** [reply]
thank u !

**Comment by huydang168tb** [comment]
![oie_ZDj2lQfWcqAR](https://github.com/supabase/supabase/assets/172629499/27b715b2-bc49-4ef4-a45c-930012e8bfbe)

**Comment by soedirgo** [comment]
Hey all! Can you share a repo where I can reproduce the error? This needs to be fixed without having to pin the version.

**Comment by soedirgo** [comment, marked as answer]
This should be fixed as of supabase-js v2.44.3.
